app/src/main/java/com/pokemonshowdown/data/MyApplication.java

```diff
@@ -388,4 +404,73 @@ public void setBattleCount(int battleCount) {
     public void setRoomCategoryList(HashMap<String, JSONArray> roomCategoryList) {
         mRoomCategoryList = roomCategoryList;
     }
+
+
+    public class UpdateCheckTask extends AsyncTask<Void, Void, Void> {
+        private final static String VERSION_URL = "http://ns3367227.ip-37-187-3.eu/showdown/version.txt";
+
+        private int status;
+        private String serverVersion = null;
+
+        @Override
+        protected Void doInBackground(Void... voids) {
+            final HttpParams httpParams = new BasicHttpParams();
+            HttpConnectionParams.setConnectionTimeout(httpParams, 50000);
+            HttpClient client = new DefaultHttpClient(httpParams);
+            HttpGet httpget = new HttpGet(VERSION_URL);
+            HttpResponse response = null;
+            try {
+                response = client.execute(httpget);
+                status = response.getStatusLine().getStatusCode();
+                if (status == HttpStatus.SC_OK) {
+                    HttpEntity entity = response.getEntity();
+                    serverVersion = EntityUtils.toString(entity);
+                }
+            } catch (IOException e) {
+                return null;
+            }
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(Void aVoid) {
+            super.onPostExecute(aVoid);
+            if (status == HttpStatus.SC_OK) {
+                try {
+                    String currentVersion = MyApplication.this.getPackageManager().getPackageInfo(MyApplication.this.getApplicationContext().getPackageName(), 0).versionName;
+                    if (isUpdateAvailable(currentVersion, serverVersion)) {
+                        LocalBroadcastManager.getInstance(MyApplication.getMyApplication()).sendBroadcast(new Intent(MyApplication.ACTION_FROM_MY_APPLICATION).putExtra(MyApplication.EXTRA_DETAILS, MyApplication.EXTRA_UPDATE_AVAILABLE));
+                    }
+                } catch (PackageManager.NameNotFoundException e) {
+                    e.printStackTrace();
```

Comment:
I didnt think it'd be necessary, the error will never happen.

i'll do it nevertheless